src/Routes.jsx

```diff
@@ -15,21 +15,51 @@ const routesLayer = new GeoJsonLayer({
   extruded: true,
 });
 
+// placed outside of component to use in both stop & vehicle layer
+const ICON_MAPPING = {
+  marker: {
+    x: 0, y: 0, width: 128, height: 128, mask: true,
+  },
+};
 
 class Routes extends Component {
+  getStopMarkers() {
+    /* returns new DeckGL Icon Layer displaying all stops on given routes */
+
+    // Push stop markers into data array
+    const data = this.props.state.routes.filter(r => r.stops).reduce((acc, curr) =>
+      acc.concat(curr.stops.reduce((a, c) => {
+        a.push({
```

Comment:
if we wanted to be more functional (like functional-programming style), we could do `return [...a, { newObject, etc }, ];` instead

Reply:
yes, this makes sense also